lib/ecto_diff.ex

```diff
@@ -367,6 +367,32 @@ defmodule EctoDiff do
     end
   end
 
+  defp diff_association(previous, current, %{cardinality: :many, through: [through, field]}, acc, opts) do
+    assoc_diff =
+      acc
+      |> Keyword.get_values(through)
+      |> List.flatten()
+      |> Enum.map(& &1[:changes][field])
+
+    diff =
+      case assoc_diff do
+        [_ | _] = diff ->
+          List.flatten(diff)
+
+        _ ->
+          {previous, current} =
+            case length(previous) - length(current) do
+              0 -> {previous, current}
+              n when n > 0 -> {previous, current ++ List.duplicate(nil, n)}
+              n when n < 0 -> {previous ++ List.duplicate(nil, -n), current}
```

Comment:
:warning: Uncovered lines. It's also hard for me to tell what this code is meant to be doing. What is the wildcard matching on? Is this meant handle the cases of added and removed associations? Can we re-use the logic from https://github.com/peek-travel/ecto_diff/blob/a59c906355fadeaa8bd51c2efeb040acd97ab311/lib/ecto_diff.ex#L396

Which handles adding and removing associations using primary keys?